examples/pointer_theme_window.rs

```diff
@@ -0,0 +1,483 @@
+    Connection, Dispatch, QueueHandle,
+};
+
+fn main() {
+    env_logger::init();
+
+    let conn = Connection::connect_to_env().unwrap();
+
+    let (globals, mut event_queue) = registry_queue_init(&conn).unwrap();
+    let qh = event_queue.handle();
+
+    let mut simple_window = SimpleWindow {
+        registry_state: RegistryState::new(&conn, &qh),
+        seat_state: SeatState::new(),
+        output_state: OutputState::new(),
+        compositor_state: CompositorState::bind(&globals, &qh)
+            .expect("wl_compositor not available"),
+        shm_state: ShmState::bind(&globals, &qh).expect("wl_shm not available"),
+        xdg_shell_state: XdgShellState::bind(&globals, &qh).expect("xdg shell not available"),
+        xdg_window_state: XdgWindowState::bind(&globals, &qh),
+
+        exit: false,
+        first_configure: true,
+        pool: None,
+        width: 256,
+        height: 256,
+        shift: None,
+        buffer: None,
+        window: None,
+        pointer_surface: None,
+        keyboard: None,
+        keyboard_focus: false,
+        pointer: None,
+        themed_pointer: None,
+    };
+
+    while !simple_window.registry_state.ready() {
+        event_queue.blocking_dispatch(&mut simple_window).unwrap();
+    }
+
+    let pool = SlotPool::new(
+        simple_window.width as usize * simple_window.height as usize * 4,
+        &simple_window.shm_state,
+    )
+    .expect("Failed to create pool");
+    simple_window.pool = Some(pool);
+
+    let window_surface = simple_window.compositor_state.create_surface(&qh).unwrap();
+    let pointer_surface = simple_window.compositor_state.create_surface(&qh).unwrap();
+
+    let window = Window::builder()
+        .title("A wayland window")
+        // GitHub does not let projects use the `org.github` domain but the `io.github` domain is fine.
+        .app_id("io.github.smithay.client-toolkit.SimpleWindow")
+        .min_size((256, 256))
+        .map(
+            &qh,
+            &simple_window.xdg_shell_state,
+            &mut simple_window.xdg_window_state,
+            window_surface,
+        )
+        .expect("window creation");
+
+    simple_window.pointer_surface.replace(pointer_surface);
+    simple_window.window = Some(window);
+
+    // We don't draw immediately, the configure will notify us when to first draw.
+
+    loop {
+        event_queue.blocking_dispatch(&mut simple_window).unwrap();
+
+        if simple_window.exit {
+            println!("exiting example");
+            break;
+        }
+    }
+}
+
+struct SimpleWindow {
+    registry_state: RegistryState,
+    seat_state: SeatState,
+    output_state: OutputState,
+    compositor_state: CompositorState,
+    shm_state: ShmState,
+    xdg_shell_state: XdgShellState,
+    xdg_window_state: XdgWindowState,
+
+    exit: bool,
+    first_configure: bool,
+    pool: Option<SlotPool>,
+    width: u32,
+    height: u32,
+    shift: Option<u32>,
+    buffer: Option<Buffer>,
+    window: Option<Window>,
```

Comment:
The simple window example probably has been modified to compensate for the registry changes, we can create the window before looping for hotpluggable globals, effectively removing the need for an Option around Buffer and window.